src/Eshopworld.Core/Events/BaseEvent.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Newtonsoft.Json;
+
+namespace Eshopworld.Core.Events
+{
+    /// <summary>
+    /// Standard interface that every event that will be published & consumed from the event streaming bus will expose
+    /// </summary>
+    public abstract class BaseEvent
```

Comment:
I was hoping we will avoid inheritance for the events and we will introduce proper envelope and payload concepts.
Internally, we should have the wrapper but externally there is no reason for the events to inherit from anything (base class or interface).

I was thinking you could have the construct like:
```
class KafkaMessage {
  public MessageEnvelope Envelope;
  public string/jobject Payload;
}

class MessageEnvelope {
  public string CorrelationId;
  // .. and so on - all the fields you want to be available
}
```

This way you can ensure/control the fact that envelope is always readonly and is responsible of the library rather than consumer of the library.

Reply:
I like your approach, the inheritance approach used is something I've done before to make sure mandatory fields on the event are set, but this will cover that usecase as well and make it less awkward. @dguerin what is your opinion?

Reply:
Yeah, I like the concept here and it's useful for the meta-data properties. However, we do need to constrain the events to the specific property as standards. How do you suggest we enforce that here or is that all something you see as being within the SDK to retrieve and append to the message?

Reply:
NMV, I see what you want to do below.

Reply:
Yes let's go with this, it's a good choice.